coap-gateway/service/service.go

```diff
@@ -211,6 +213,14 @@ func New(ctx context.Context, config Config, logger *zap.Logger) (*Service, erro
 		}
 	}
 
+	provider, err := provider.NewPlgdProvider(config.OAuthDeviceClient,
+		logger,
+		config.Clients.AuthServer.OwnerClaim,
+		"query", "offline", "code")
+	if err != nil {
+		return nil, fmt.Errorf("cannot create device provider: %w", err)
```

Comment:
add `resourceSubscriber.Close()` before you call return. it is free all allocated resources before.